posthog/temporal/data_imports/pipelines/mssql/mssql.py

```diff
@@ -262,6 +267,75 @@ def _get_table_chunk_size(
     return min_chunk_size
 
 
+def _get_table_stats(cursor: Cursor, schema: str, table_name: str) -> tuple[int, float]:
+    """Calculate the number of rows and size of a table.
+
+    Uses sp_spaceused stored procedure which is the official way to get accurate table size and row count.
+    Falls back to simpler version for SQL Server versions before 2012.
+    """
+    # Try modern version first (SQL Server 2012+)
+    query = "EXEC sp_spaceused %(full_table_name)s, @updateusage = 'TRUE'"
+
+    try:
+        cursor.execute(query, {"full_table_name": f"[{schema}].[{table_name}]"})
+    except Exception:
+        # If @updateusage parameter fails, try the older version
+        query = "EXEC sp_spaceused %(full_table_name)s"
+        cursor.execute(query, {"full_table_name": f"[{schema}].[{table_name}]"})
+
+    result = cursor.fetchone()
+    if result is None:
+        raise ValueError("_get_partition_settings: sp_spaceused returned no results")
+
+    # sp_spaceused returns: name, rows, reserved, data, index_size, unused
+    _, total_rows, _, data_size, _, _ = result
+
+    # Convert string values to numbers
+    total_rows = int(total_rows)
+
+    # Parse size with unit (e.g. "1024.45 MB" -> 1024.45, "MB")
+    size_parts = data_size.strip().split(" ")
+    if len(size_parts) != 2:
+        raise ValueError(
+            f"_get_partition_settings: Invalid sp_spaceused result: expected 2 parts, got {len(size_parts)}"
+        )
+
+    size_value = float(size_parts[0])
+    unit = size_parts[1].upper()
+
+    # Convert to bytes based on unit
+    multiplier = {"KB": 1024, "MB": 1024 * 1024, "GB": 1024 * 1024 * 1024, "TB": 1024 * 1024 * 1024 * 1024}.get(unit)
```

Comment:
Can it ever return just bytes (e.g. < 1kb with no unit)?

Reply:
I don't think so, at least from my testing. Seems like the minimum page size is 8KB, so it will increment in 8KB steps.

Reply:
Lovely - 🚢